### ISSUE TITLE
Elm 0.18 Support

### ISSUE THREAD

**Issue by carleryd** [jahewson/elm-graphql#36]
https://github.com/elm-lang/elm-platform/blob/master/upgrade-docs/0.18.md

**Comment by TimPerry** [comment]
+1 and update on this?

**Comment by jahewson** [comment]
Yep, after much delay I'm working on this now. Having to rewrite the decoder codegen as it relies heavily on a custom `infix` operator.

**Comment by Bellk17** [comment]
@jahweson my fork is 0.18 compliant, though using a different `GraphQl.elm` file.

**Comment by TimPerry** [comment]
@Bellk17 Any change you could do a PR?

**Comment by Bellk17** [comment]
@TimPerry I'll look into it tonight. I've made a number of changes on my fork to utilize this on a build server, so need to make sure no breaking changes.

**Comment by maticzav** [comment]
Any progress with this?

**Comment by Gregoirevda** [comment]
+1 any update?

**Comment by nicu-chiciuc** [comment]
It seems like the last commit was made a year ago. Is the project abandoned?

**Comment by nickdima** [comment]
This one seems more active: https://github.com/jamesmacaulay/elm-graphql

**Comment by dillonkearns** [comment]
Hello @jahewson, I [recently published a library called Graphqelm](https://discourse.elm-lang.org/t/introducing-graphqelm-a-tool-for-type-safe-graphql-queries/472) that I think achieves similar goals to this one, and has support for Elm 0.18. It actually takes the approach of generating functions for the entire schema which guarantee that you only build valid, type-safe queries. I recommend that anybody on this thread check it out! Here's a [live Ellie demo](https://rebrand.ly/graphqelm).

I believe that with `graphqelm`, the technique of generating code for specific queries is obsolete (though perhaps you might disagree, or there is a use case I am missing). I'd be curious to hear everyone's thoughts on this.

**Comment by Gregoirevda** [comment]
Will check that out!

**Comment by daniel-kun** [comment]
IMHO it sounds reasonable to generate the whole scheme (except when it might be gigantic). Will definitely try it out when I want to access a GraphQL API from Elm again. Thanks!

**Comment by dillonkearns** [comment]
@daniel-kun I agree. I've been using the Github API as an example. It generates a total of 308 files, but it doesn't seem to be an issue. It's in the [examples/src/Github.elm](https://github.com/dillonkearns/graphqelm/blob/master/examples/src/Github.elm) file in the repo.

**Comment by dillonkearns** [comment]
And another nice thing is that it gives you a clear separation between the generated code and your code, so you can freely edit your code without worrying about breaking something that was auto-generated. So having lots of files isn't necessarily an issue unless it slows down compile times or increases bundle sizes too much. That hasn't been an issue with the Github example, and Elm 0.19 is going to improve dead code elimination and some compile performance issues I believe.
